cmd/openshift-install/create.go

```diff
@@ -525,58 +523,31 @@ func waitForBootstrapConfigMap(ctx context.Context, client *kubernetes.Clientset
-// before continuing.
-func waitForStableSNOBootstrap(ctx context.Context, config *rest.Config) error {
+// If the bootstrap etcd member is cleaned up before it has been removed from the etcd cluster, the
+// etcd cluster cannot maintain quorum through the rollout of any single permanent member.
+func waitForEtcdBootstrapMemberRemoval(ctx context.Context, config *rest.Config) error {
 	timeout := 5 * time.Minute
 
-	// If we're not in a single node deployment, bail early
-	if isSNO, err := IsSingleNode(); err != nil {
-		logrus.Warningf("Can not determine if installing a Single Node cluster, continuing as normal install: %v", err)
-		return nil
-	} else if !isSNO {
-		return nil
-	}
-
-	snoBootstrapContext, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
-
 	untilTime := time.Now().Add(timeout)
 	timezone, _ := untilTime.Zone()
-	logrus.Info("Detected Single Node deployment")
 	logrus.Infof("Waiting up to %v (until %v %s) for the bootstrap etcd member to be removed...",
 		timeout, untilTime.Format(time.Kitchen), timezone)
 
-	client, err := dynamic.NewForConfig(config)
+	client, err := operatorclient.NewForConfig(config)
 	if err != nil {
-		return fmt.Errorf("error creating dynamic client: %w", err)
+		return fmt.Errorf("error creating operator client: %w", err)
 	}
-	gvr := schema.GroupVersionResource{
-		Group:    operatorv1.SchemeGroupVersion.Group,
-		Version:  operatorv1.SchemeGroupVersion.Version,
-		Resource: "etcds",
-	}
-	resourceClient := client.Resource(gvr)
 	// Validate the etcd operator has removed the bootstrap etcd member
-	return wait.PollUntilContextCancel(snoBootstrapContext, 1*time.Second, true, func(ctx context.Context) (done bool, err error) {
-		etcdOperator := &operatorv1.Etcd{}
-		etcdUnstructured, err := resourceClient.Get(ctx, "cluster", metav1.GetOptions{})
+	return wait.PollUntilContextTimeout(ctx, 1*time.Second, timeout, true, func(ctx context.Context) (done bool, err error) {
+		etcd, err := client.OperatorV1().Etcds().Get(ctx, "cluster", metav1.GetOptions{})
 		if err != nil {
-			// There might be service disruptions in SNO, we log those here but keep trying with in the time limit
-			logrus.Debugf("Error getting ETCD Cluster resource, retrying: %v", err)
+			logrus.Debugf("Error getting etcd operator singleton, retrying: %v", err)
 			return false, nil
 		}
-		err = runtime.DefaultUnstructuredConverter.FromUnstructured(etcdUnstructured.Object, etcdOperator)
-		if err != nil {
-			// This error should not happen, if we do, we log the error and keep retrying until we hit the limit
-			logrus.Debugf("Error parsing etcds resource, retrying: %v", err)
-			return false, nil
-		}
-		for _, condition := range etcdOperator.Status.Conditions {
+
+		for _, condition := range etcd.Status.Conditions {
 			if condition.Type == "EtcdBootstrapMemberRemoved" {
-				return configv1.ConditionStatus(condition.Status) == configv1.ConditionTrue, nil
```

Comment:
Reference for where we set this in the CEO, right after the bootstrap member removal: 
https://github.com/openshift/cluster-etcd-operator/blob/658cb7e720e286973bfb5aacc3a371d2cf65b229/pkg/operator/bootstrapteardown/bootstrap_teardown_controller.go#L144-L160

Also side note, but right before that we have the check for the bootstrap completion configmap via the library-go helper `IsBootstrapComplete()` with the comment:
```
In HA clusters, the bootstrap node is torn down as soon as the configmap is created with the appropriate content.
```
https://github.com/openshift/library-go/blob/64d8d9e8080894cc802cff2133bd11fe7f191d2d/pkg/operator/bootstrap/bootstrap.go#L22-L24

Once this goes in, I'm guessing we can update that to say otherwise, i.e additional conditions apply before the node is torn down.